lib/cli/run-helpers.js

```diff
@@ -140,9 +157,27 @@ const singleRun = async (mocha, {exit}, fileCollectParams) => {
  * @private
  */
 const parallelRun = async (mocha, options, fileCollectParams) => {
-  const files = collectFiles(fileCollectParams);
-  debug('executing %d test file(s) in parallel mode', files.length);
-  mocha.files = files;
+  const fileCollectionObj = collectFiles(fileCollectParams);
+
+  if (fileCollectionObj.unmatchedFiles.length > 0) {
+    fileCollectionObj.unmatchedFiles.forEach(({pattern, absolutePath}) => {
+      console.error(
+        ansi.yellow(
+          `Warning: Cannot find any files matching pattern "${pattern}" at the absolute path "${absolutePath}"`
+        )
+      );
+    });
+    console.log(
+      'No test file(s) found with the given pattern, exiting with code 1'
+    );
+    return mocha.run(exitMocha(1));
+  }
```

Comment:
[Refactor] This is about the same as the previous block. Up for extracting out a function?

Having two areas of code that are roughly the same isn't the worst thing in the world, but given how they're pretty much exactly identical - I think it'd make the code more readable & easy to understand. Plus if we have to change it later, it'll be easy to forget one.

Reply:
Yeah that makes sense to me 👍🏻. I extracted the common code to a function `handleUnmatchedFiles` that is now called both by `singleRun` and `parallelRun` instead of the previous duplication of code.